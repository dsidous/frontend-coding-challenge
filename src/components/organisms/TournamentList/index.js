import React from 'react';
import styled from 'styled-components';

import theme from '../../../theme';
import Tournament from '../../molecules/Tournament';
import Loading from '../../atoms/Loading';

const StyledDiv = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
  grid-gap: ${theme.spacing(6)};
`;

const TournamentList = ({ data }) => {
  if (data.error) {
    return <div>Something went wrong.</div>;
  }

  if (data.loading) {
    return <Loading />;
  }

  return (
    <StyledDiv>
      {data.tournaments &&
        data.tournaments.map(tournament => (
          <Tournament tournament={tournament} key={tournament.id} />
        ))}
    </StyledDiv>
  );
};

export default TournamentList;
