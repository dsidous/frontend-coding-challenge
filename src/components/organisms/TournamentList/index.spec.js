import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import TournamentList from './';

const mockDispatch = jest.fn();

jest.mock('react-redux', () => ({
  useDispatch: () => mockDispatch
}));

describe('TournamentList', () => {
  let data;
  mockDispatch.mockReturnValue(data);

  beforeEach(() => {
    data = {
      tournaments: [
        {
          id: 'mock id',
          name: 'mock name',
          organizer: 'mock organizer',
          game: 'mock game',
          participants: {
            current: 1,
            max: 2
          },
          startDate: '2020.12.12 10:10:10'
        },
        {
          id: 'mock id2',
          name: 'mock name',
          organizer: 'mock organizer',
          game: 'mock game',
          participants: {
            current: 1,
            max: 2
          },
          startDate: '2020.12.12 10:10:10'
        }
      ]
    };
  });

  it('Should render as expected', () => {
    const { asFragment } = render(<TournamentList {...{ data }} />);
    expect(asFragment).toMatchSnapshot();
  });

  it('Should show the loading screen', () => {
    data.loading = true;
    const { getByText } = render(<TournamentList {...{ data }} />);

    expect(getByText('Loading Tournaments...')).toBeTruthy();
  });

  it('Should show the error screen', () => {
    data.error = true;
    const { getByText } = render(<TournamentList {...{ data }} />);

    expect(getByText('Something went wrong')).toBeTruthy();
  });
});
