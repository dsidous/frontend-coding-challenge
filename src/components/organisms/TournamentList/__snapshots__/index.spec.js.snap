// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TournamentList Should render as expected 1`] = `[Function]`;
